src/lib.rs

```diff
@@ -275,6 +302,20 @@ async fn get_intensity_for_url(url: &str) -> Result<i32> {
     Ok(intensity)
 }
 
+/// Retrieves the intensity value from a structure
+async fn get_intensity_for_url_national(url: &str) -> Result<i32> {
+    let result = get_response::<DataVec>(url).await?;
+
+    let intensity = result
+        .data
+        .first()
+        .ok_or_else(|| ApiError::Error("No data found".to_string()))?
+        .intensity
+        .forecast;
```

Comment:
I notice that for the national data the responses have an `actual` value, which presumably it's what we'd want to return instead?

But I'm fairly new and I may miss something. In my mind the forecast was a "guess" of what the value would be for that region/datetime range - while the `actual` is the actual value measured?

Annoyingly it seems like the response for region/postcode doesn't have it: However this shouldn't be a big deal as we can add the `actual` field to `Intensity` as an Option, e.g.

```Rust
pub struct Intensity {
    forecast: i32,
    actual: Option<i32>,
    index: String,
}
```

and serde shouldn't have much trouble deserialising the response regardless of whether the `actual` field is there (national) or not (region/postcode).

But maybe I'm missing something :)

Reply:
no, you are spot on. I overlooked the difference in the data returned by national vs regional. From what I heard the regional data is far less reliable than the national one so maybe they used `forecast` as it is more guesswork than facts.
It also means we need to use actual when available and forecast when not.